lib/anycablebility/cli/app.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'logger'
+require 'anycablebility/rpc'
+require 'anycablebility/tests'
+require 'anycablebility/client_factory'
+
+module Anycablebility
+  module Cli
+    # This class is the entry point of Anycablebility
+    class App
+      def run(
+        target: 'ws://0.0.0.0:8080/cable',
+        redis: 'redis://localhost:6379',
+        debug: false
+      )
+        logger = Logger.new(STDOUT)
+        logger.level = debug ? Logger::DEBUG : Logger::WARN
+
+        rpc = Rpc.instance.configure(redis, debug, logger)
```

Comment:
The purpose of the global/shared logger is to get rid of these parameters: `debug`, `logger`.

Let's do the following:

- introduce a `Logging` module:

```ruby
module Anycablebility
  module Logging
    PREFIX = 'AC'
    
    class << self
      attr_writer :logger
    end

    def logger
      Anycablebility.logger
    end
    
   # helper method
   # later we'll be able to add a colorising here, depending on the level
    def log(level, message)
      return unless logger
      logger.send(level.to_sym, PREFIX) { message }
    end
  end
end
```

- Then in `#run` method:

```ruby
Anycablebility.logger = Logger.new(STDOUT).tap do |logger|
  logger.level = debug ? Logger::DEBUG : Logger::WARN
end
```

- When configuring AnyCable:

```ruby
Anycable.logger = Anycablebility.logger
```

- And anywhere you need a logger:

```ruby
include Logging
```